scylla-macros/src/serialize/row.rs

```diff
@@ -205,11 +205,30 @@ impl Generator for ColumnSortingGenerator<'_> {
             struct_name.span(),
         );
         let mut partial_generics = self.ctx.generics.clone();
-        let partial_lt: syn::LifetimeParam = syn::parse_quote!('scylla_ser_partial);
+        let partial_lt: syn::Lifetime = syn::parse_quote!('scylla_ser_partial);
+
         if !self.ctx.fields.is_empty() {
+            // all fields have to outlive the partial struct lifetime
+            partial_generics
+                .params
+                .iter_mut()
+                .filter_map(|p| match p {
+                    syn::GenericParam::Lifetime(lt) => Some(lt),
+                    _ => None,
+                })
+                .for_each(|lt| {
+                    lt.bounds.push(partial_lt.clone());
+                });
+
+            // now add the partial struct lifetime
             partial_generics
                 .params
-                .push(syn::GenericParam::Lifetime(partial_lt.clone()));
+                .push(syn::GenericParam::Lifetime(syn::LifetimeParam {
+                    attrs: vec![],
+                    lifetime: partial_lt.clone(),
+                    colon_token: None,
+                    bounds: syn::punctuated::Punctuated::new(),
+                }));
```

Comment:
:question: I feel this commit requires a more comprehensive explanation in the commit message. An example there and here in a comment would definitely help, too.